src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -158,6 +158,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
+        $choiceTranslationDomain = $options['choice_translation_domain'];
+        $translationDomain = $options['translation_domain'];
+
+        if ($view->parent) {
+            if (null === $choiceTranslationDomain) {
+                $choiceTranslationDomain = $view->parent->vars['translation_domain'];
```

Comment:
if you pass ``null``, you need to read it from ``translation_domain`` (and do it *after* it has itself be inherited eventually)